src/gt4py/next/ffront/decorator.py

```diff
@@ -218,6 +219,13 @@ def with_backend(self, backend: ppi.ProgramExecutor) -> Program:
     def with_grid_type(self, grid_type: GridType) -> Program:
         return dataclasses.replace(self, grid_type=grid_type)
 
+    def with_bound_args(self, **kwargs) -> ProgramWithBoundArgs:
+        for key in kwargs.keys():
+            if all(key != param.id for param in self.past_node.params):
+                raise RuntimeError(f"Keyword argument `{key}` not among program inputs.")
+
+        return ProgramWithBoundArgs(self.past_node, self.closure_vars, bound_args=kwargs)
```

Comment:
Please use a way that is generic in the sense that you do not have to change this method when the list of attributes of this class changes. In this case for example `self.backend` has not been forwarded and depending on what attributes are missing this can lead to strange behavior. Something along the lines of (not sure about the interface to get the field information):
```suggestion
        return ProgramWithBoundArgs(bound_args=kwargs, **{field.name: getattr(self, field.name) for field in dataclasses.fields(self)})
```